hazelcast/src/main/java/com/hazelcast/internal/monitors/SlowOperationPlugin.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.monitors;
+
+import com.hazelcast.instance.GroupProperties;
+import com.hazelcast.internal.management.dto.SlowOperationDTO;
+import com.hazelcast.internal.management.dto.SlowOperationInvocationDTO;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.spi.impl.NodeEngineImpl;
+import com.hazelcast.spi.impl.operationservice.InternalOperationService;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.util.List;
+
+import static com.hazelcast.instance.GroupProperty.PERFORMANCE_MONITOR_SLOW_OPERATIONS_PERIOD_SECONDS;
+import static com.hazelcast.instance.GroupProperty.SLOW_OPERATION_DETECTOR_ENABLED;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * A {@link PerformanceMonitorPlugin} that displays the slow executing operations. For more information see
+ * {@link com.hazelcast.spi.impl.operationexecutor.slowoperationdetector.SlowOperationDetector}.
+ */
+public class SlowOperationPlugin extends PerformanceMonitorPlugin {
+
+    private final InternalOperationService operationService;
+    private final long periodSeconds;
+    private final ILogger logger;
+
+    public SlowOperationPlugin(NodeEngineImpl nodeEngine) {
+        this.logger = nodeEngine.getLogger(SlowOperationPlugin.class);
+        this.operationService = nodeEngine.getOperationService();
+        this.periodSeconds = getPeriodSeconds(nodeEngine);
+    }
+
+    private long getPeriodSeconds(NodeEngineImpl nodeEngine) {
+        GroupProperties groupProperties = nodeEngine.getGroupProperties();
+        if (!groupProperties.getBoolean(SLOW_OPERATION_DETECTOR_ENABLED)) {
+            return -1;
```

Comment:
I think the disabled magic number was `0`, not `-1`. Maybe another good candidate for a constant.